src/hevm/src/EVM/ABI.hs

```diff
@@ -537,6 +540,28 @@ listP parser = between (char '[') (char ']') ((do skipSpaces
                                                   return a) `sepBy` (char ','))
 
 
+-- TODO: teach this to handle complex types for symbolic buffers
+decodeBuffer :: [AbiType] -> Buffer -> Maybe (Either [AbiValue] [SWord 256])
```

Comment:
The return type is a little gross here, dunno if there is a cleaner way... :thinking:

Reply:
do we really need to special case the concrete case here? Can't we just have:
` decodeBuffer :: [AbiType] -> Buffer -> Maybe [SWord 256]`?

Reply:
If one needs concreteness that can be taken into account at the callsite. Current uses all do the same thing regardless of concreteness

Reply:
would also be nice to use this to clean up `getStaticAbiArgs` of `test.hs`